python/graphstorm/config/argument.py

```diff
@@ -76,7 +76,35 @@
 ]
 
 def get_argument_parser():
-    """Parse command-line arguments"""
+    """ Get GraphStorm CLIs argument parse.
+
+    This argument parser can accept and parse all GraphStorm model training and inference configurations
+    defined in a yaml file. It also can accept and parse the corresponding arugments in
+    GraphStorm launch CLIs. Specifically, it will parses yaml config file first, and then parses
+    arguments to overwrite parameters defined in the yaml file or add new parameters.
+
+    Examples:
+    ----------
+
+    .. code:: python
+
+        from graphstorm.config import get_argument_parser
+        
+        if __name__ == '__main__':
+            # use GraphStorm argument parser to accept configuration yaml file and other arguments
+            arg_parser = get_argument_parser()
+       
+            # parse all arguments and split GraphStorm's built-in arguments from the customized ones
+            gs_args, unknown_args = arg_parser.parse_known_args()
+       
+            print(f'GS arguments: {gs_args}')
+            print(f'Non GS arguments: {unknown_args}')
```

Comment:
Shall we also demostrate how to create a GSConfig object?

Reply:
Yes. And then we need to release the GSConfig API too. Let me create another PR to release it.